crates/matrix-sdk-crypto/src/machine/mod.rs

```diff
@@ -2378,6 +2378,35 @@ impl OlmMachine {
         DehydratedDevices { inner: self.to_owned() }
     }
 
+    /// Get the cached dehydrated device pickle key if any.
+    ///
+    /// None if the key was not previously cached (via
+    /// [`Self::save_dehydrated_device_pickle_key`]).
+    ///
+    /// Should be used to periodically rotate the dehydrated device to avoid
+    /// one-time keys exhaustion and accumulation of to_device messages.
+    pub async fn get_dehydrated_device_pickle_key(
+        &self,
+    ) -> StoreResult<Option<DehydratedDeviceKey>> {
+        self.inner.store.load_dehydrated_device_pickle_key().await
+    }
+
+    /// Store the dehydrated device pickle key in the crypto store.
+    ///
+    /// This is useful if the client wants to periodically rotate dehydrated
+    /// devices to avoid one-time keys exhaustion and accumulated to_device
+    /// problems.
+    pub async fn save_dehydrated_device_pickle_key(
+        &self,
+        dehydrated_device_pickle_key: &DehydratedDeviceKey,
+    ) -> Result<(), CryptoStoreError> {
+        let changes = Changes {
+            dehydrated_device_pickle_key: Some(dehydrated_device_pickle_key.clone()),
+            ..Default::default()
+        };
+        self.inner.store.save_changes(changes).await
+    }
```

Comment:
All the methods related to dehydrated devices are under the "fake", "API" struct called `DehydratedDevices` let's move these under there as well. We should try really hard to avoid adding new methods directly under the `OlmMachine`.

Reply:
Moved https://github.com/matrix-org/matrix-rust-sdk/pull/4383/commits/3d32679d1b10d4bed8eecd87d6de00e6fbadb673